### PR TITLE
Sample MQTT receiver

### PR DESCRIPTION
**Pull request by Meruja**
## Purpose
To demonstrates how to configure WSO2 Stream Processor to send events via MQTT transport in xml format ,and view the output on the mqtt-consumer. But the mqtt-consumer is not available in sample-clients of product-sp.
And also user can override non-optional properties( url, topic and type of mapping).
 
## Learning
Existing clients in product-sp.

### CODE REVIEW

File: modules/samples/sample-clients/pom.xml [hunk 1]
```diff
@@ -26,6 +26,13 @@
 
     <modelVersion>4.0.0</modelVersion>
     <name>WSO2 Stream Processor - Sample Clients</name>
+    <dependencies>
+        <dependency>
+            <groupId>org.apache.log4j.wso2</groupId>
+            <artifactId>log4j</artifactId>
+            <version>1.2.17.wso2v1</version>
```
Comment (mohanvive) [inline modules/samples/sample-clients/pom.xml]: Move dependency to parent pom.

File: modules/samples/sample-clients/rabbitmq-consumer/build.xml [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+            <include name="slf4j.api_*.jar"/>
+            <include name="siddhi-*.jar"/>
+            <include name="disruptor_*.jar"/>
+            <include name="org.eclipse.osgi_*.jar"/>
+            <include name="quartz_*.jar"/>
+            <include name="antlr4-runtime_*.jar"/>
+            <include name="io.dropwizard.metrics.core_*.jar"/>
+            <include name="com.google.*.jar"/>
+            <include name="org.apache.ws.commons.axiom.axiom-api_*.jar"/>
+            <include name="jaxen_*.jar"/>
```
Comment (mohanvive) [inline modules/samples/sample-clients/rabbitmq-consumer/build.xml]: Whether you required all these dependencies?

File: modules/samples/sample-clients/mqtt-consumer/pom.xml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  ~ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  ~ KIND, either express or implied.  See the License for the
+  ~ specific language governing permissions and limitations
+  ~ under the License.
+  -->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <groupId>org.wso2.sp</groupId>
+        <artifactId>sample-clients</artifactId>
+        <version>4.0.0-beta2-SNAPSHOT</version>
```
Comment (mohanvive) [inline modules/samples/sample-clients/mqtt-consumer/pom.xml]: Parent version is changed now. Please re-check.

File: modules/samples/sample-clients/rabbitmq-consumer/pom.xml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  ~  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  ~  KIND, either express or implied.  See the License for the
+  ~  specific language governing permissions and limitations
+  ~  under the License.
+  -->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <groupId>org.wso2.sp</groupId>
+        <artifactId>sample-clients</artifactId>
+        <version>4.0.0-beta2-SNAPSHOT</version>
```
Comment (mohanvive) [inline modules/samples/sample-clients/rabbitmq-consumer/pom.xml]: Parent pom version is changed. Please re-check.

File: modules/samples/sample-clients/mqtt-consumer/pom.xml (hunk 3 above)
Comment (Meruja) [reply modules/samples/sample-clients/mqtt-consumer/pom.xml]: ok anna.

Comment (Meruja) [reply modules/samples/sample-clients/mqtt-consumer/pom.xml]: corrected

File: modules/samples/sample-clients/rabbitmq-consumer/pom.xml (hunk 4 above)
Comment (Meruja) [reply modules/samples/sample-clients/rabbitmq-consumer/pom.xml]: corrected

File: modules/samples/sample-clients/pom.xml (hunk 1 above)
Comment (mohanvive) [reply modules/samples/sample-clients/pom.xml]: @Meruja Please fix ..

Comment (Meruja) [reply modules/samples/sample-clients/pom.xml]: sorry for the delay. moved to parent

File: modules/samples/sample-clients/rabbitmq-consumer/build.xml (hunk 2 above)
Comment (Meruja) [reply modules/samples/sample-clients/rabbitmq-consumer/build.xml]: yeah

File: modules/samples/sample-clients/rabbitmq-consumer/build.xml [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+    <target name="compile" depends="init">
+        <javac srcdir="${src.dir}" destdir="${class.dir}" compiler="modern">
+            <include name="*/**"/>
+            <classpath refid="javac.classpath"/>
+        </javac>
+    </target>
+
+    <target name="run" depends="compile">
+        <echo>Publishing events to RabbitMQ endpoints to port 5672</echo>
```
Comment (ramindu90) [inline modules/samples/sample-clients/rabbitmq-consumer/build.xml]: Can you add the ant command description as https://github.com/ramindu90/product-sp/blob/master/modules/samples/sample-clients/kafka-producer/build.xml#L84

File: modules/samples/sample-clients/rabbitmq-consumer/pom.xml [hunk 6]
```diff
@@ -0,0 +1,51 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~  Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (ramindu90) [inline modules/samples/sample-clients/rabbitmq-consumer/pom.xml]: license header should be 2017

File: modules/samples/sample-clients/rabbitmq-consumer/pom.xml [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+    <modelVersion>4.0.0</modelVersion>
+    <name>WSO2 Stream Processor - Sample - RabbitMQ Consumer</name>
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <configuration>
+                    <source>1.7</source>
```
Comment (ramindu90) [inline modules/samples/sample-clients/rabbitmq-consumer/pom.xml]: can we change this to 1.8?

File: modules/samples/sample-clients/rabbitmq-consumer/build.xml (hunk 5 above)
Comment (Meruja) [reply modules/samples/sample-clients/rabbitmq-consumer/build.xml]: ok sure ayye

**Pull request merged**